### PR TITLE
Enable intermittently failed dynamic listener test

### LINKED ISSUE #36929: Enable `testMultipleDynamicListenersWithAsyncCall` jballerina integration test case
**Description:**
Above test case is disabled with #  due to intermittent failures. 

**Describe your task(s)**

**Related Issues (optional):**

**Suggested Labels (optional):**

**Suggested Assignees (optional):**

### PR DESCRIPTION
**Pull request by gabilang**
Fixes #36929

## Approach
Revamp existing test

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [x] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: tests/jballerina-integration-test/src/test/resources/packaging/dynamic_listener_async_call_test_project/main.bal [hunk 1]
```diff
@@ -62,40 +81,30 @@ public class ListenerObject2 {
...
-    runtime:registerListener(lo2);
-    runtime:registerListener(lo3);
-    runtime:sleep(2);
-    runtime:deregisterListener(ep);
-    runtime:deregisterListener(lo2);
-    runtime:deregisterListener(lo3);
+public function incrementCount() {
+    count += 1;
+}
+public function assertCount(int val) {
```
Comment (Nadeeshan96) [inline tests/jballerina-integration-test/src/test/resources/packaging/dynamic_listener_async_call_test_project/main.bal:108]:
```suggestion
}

public function assertCount(int val) {
```

**Review by HindujaB — APPROVED** [review]
LGTM

**Comment by HindujaB** [comment]
@gabilang As discussed offline, shall we verify this test locally with multiple iterations?

**Comment by gabilang** [comment]
> @gabilang As discussed offline, shall we verify this test locally with multiple iterations?

Tested locally with few iterations. No intermittent failures observed.

**Review by Nadeeshan96 — APPROVED** [review]
LGTM

**Review by warunalakshitha — APPROVED** [review]
LGTM

**Pull request merged**